testsuite/classlibrary/TestNodeProxy.sc

```diff
@@ -40,4 +40,103 @@ TestNodeProxy : UnitTest {
 		this.assertEquals(proxy.isPlaying, false, "Setting the proxy's source should not set isPlaying = true");
 	}
 
+	test_asCode_basic {
+		var codeString = "a = NodeProxy.new(Server.fromName( 'TestNodeProxy' ));\n";
+
+		this.assertEquals(proxy.asCode.postcs, codeString,
+			"asCode-posting basic nodeproxy should post valid source code.",
+			false);
+	}
+
+	test_asCode_single {
+		var codeString = "a = NodeProxy.new(Server.fromName( 'TestNodeProxy' )).source_({ DC.ar });\n";
+		proxy.source = { DC.ar };
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting single-source nodeproxy should post simple source form.",
+			false
+		);
+	}
+
+	test_asCode_multi {
+		var asCodeString =
+		"(
+a = NodeProxy.new;
+a[5] = { DC.ar };
+a[10] = { DC.ar(0.01) };
+
+);
+";
+		proxy[5] = { DC.ar };
+		proxy[10] = { DC.ar(0.01) };
+		this.assertEquals(proxy.asCode.postcs, asCodeString,
+			"asCode-posting multi-source nodeproxy asCode should post all its sources.",
+			false
+		);
+	}
+
+	test_asCode_settings {
+		var codeString =
+		"(\n"
+		"a = NodeProxy.new(Server.fromName( 'TestNodeProxy' ));\n"
+		"a.set('freq', 440);\n"
+		");\n";
+		proxy.set('freq', 440);
+
+		this.assertEquals(proxy.asCode, codeString,
+			"asCode-posting nodeproxy with settings should post these correctly."
+		);
+	}
+
+	// this one needs the server booted ...
+	test_asCode_playState {
```

Comment:
output from this test on my machine ends with:

```
UNIT TESTS FOR 'TestNodeProxy' COMPLETED
There were no failures
FAILURE IN SERVER /n_set Node 1001 not found
FAILURE IN SERVER /n_free Node 1001 not found
```

Reply:
yes, suppressed now.